### PR TITLE
Issue #1662 - SIZE_LIMIT in MiB instead of Bytes

### PR DESCRIPTION
**Pull request by JordanShaak**
I'm having a really difficult time figuring out how I can contribute to this project. This should close issue #1662 and hopefully nothing else gets sucked into it.

### CODE REVIEW

**Comment by JordanShaak** [comment]
I don't know what to do about that failed check. I don't know where to ask anyone for help.

**Comment by mcomella** [comment]
Hey Jordan. Thanks for stopping by and helping out – we really appreciate it! :)

> I don't know where to ask anyone for help.

Sorry we missed you on IRC! We're typically around during PST and ~GMT working hours (9-5 M-F) if you want to chat in real time, or you can comment in an issue or PR directly. pocmo, liuche, and myself (mcomella) are the core maintainers if you want notify us directly.

> I'm having a really difficult time figuring out how I can contribute to this project.

Thanks for the feedback – I'm sorry that our docs aren't as good as they can be. Looking at IRC and your comments here, it looks like your concerns are 1) we don't explain well how we use git, 2) we don't explain how to set up Android Studio, and 3) we don't have documentation about how to interpret our failed checks.

Is there any other feedback you'd like to provide? I've made a note of these and we'll try to improve them in the future.

> I don't know what to do about that failed check.

Looking into this, it looks like the failure is unrelated to your patch – unfortunately, we've been having an intermittent issue with our TaskCluster job.

To get details on this failure, you can open the task, click "Focus for Android - Build - Pull Request", click "Run Logs", click "public/logs/live.log" from the dropdown - which will take you [to these logs](https://tools.taskcluster.net/groups/ZICoYCm8R0aVP4Hz4C-W_w/tasks/ZICoYCm8R0aVP4Hz4C-W_w/runs/0/logs/public%2Flogs%2Flive.log), and scroll to the bottom to see the error. In this case:

> * What went wrong:
Execution failed for task ':app:testFocusWebviewCoverageUnitTest'.
> Could not generate test report to '/opt/focus-android/app/build/reports/tests/testFocusWebviewCoverageUnitTest'.

Which is our test harness saying it couldn't generate a test report (for some reason).

Since this is unrelated to your patch, we don't need to worry about the failed check.

---

I'll be sure to review your patch by the end of the day – thanks for helping out, Jordan! :)

**Comment by mcomella** [comment]
By the way, it looks like you've gotten most everything right so far. The contributor workflow generally looks like:
- Comment in the issue you want to work on to let us know you're working on it
- Submit a pull request with your changes

Your pull request looks correctly formatted, with the exception of the merge commit. I think this may have happened if you tried to update your code onto the latest master. To update to the latest master (also known as upstream), I usually use the `git rebase` command, which will take your commits and re-layer them on top of master. On the other hand, you could also use `git merge` (or `git pull`), which is taking a direct diff of the changes together and creating a new commit on top (the merge commit).

Merge commits generally make the history hard to follow and can be avoided with `git rebase` so we use that where possible.

In your case, I think when we rebase before landing your commits, the merge commit should go away so I wouldn't worry about it too much.

File: tools/metrics/apk_size.py [hunk 1]
```diff
@@ -7,8 +7,9 @@
 from os import path, listdir, stat
 from sys import exit
 
-SIZE_LIMIT = 5242880
+SIZE_LIMIT = 5
```
Comment (mcomella) [inline tools/metrics/apk_size.py]: nit: we should specify units here to make sure no one uses the wrong ones, e.g. `SIZE_LIMIT_MB`.

Then `BYTE_LIMIT` should match a similar pattern. Maybe, `SIZE_LIMIT_BYTES`.

**Comment by JordanShaak** [comment]
> 1) we don't explain well how we use git

I've been wanting to try to get involved in open source projects for a long time, but this has been one of the biggest barriers to entry. The 'new to git' guides don't seem to answer my questions (that or I'm so unfamiliar with it that I don't understand it answering my questions), and every project I've looked into already assumes I'm familiar with git and it's been pretty tough to get involved.

> 2) we don't explain how to set up Android Studio

I have Android Studio running (I made a simple app that I use somewhat regularly), but I can't build Firefox Focus. Well, I *can* build it but it know it isn't building correctly because I can't scroll up without refreshing the page. The version I download off the Play Store doesn't have that problem so I know I'm building the app wrong but I can't figure out how to fix that.

So is my next step to just try to figure out how to merge the master branch into my fork without messing everything up? I'm not sure my git is set up correctly that I can get things from the source github...

**Comment by mcomella** [comment]
> The 'new to git' guides don't seem to answer my questions (that or I'm so unfamiliar with it that I don't understand it answering my questions), and every project I've looked into already assumes I'm familiar with git and it's been pretty tough to get involved.

Unfortunately, I learned from others so I don't have a good "new to git" guide. If you can be around during our working hours, I'd suggest asking around in #focus on IRC about your git questions.

One resource that I know is good, however, is really comprehensive so perhaps not the best for beginners is [the free git book](https://git-scm.com/book/en/v2). Github may also have some useful docs.

Another tip I'd suggest is just playing around with it! You can create a separate git repository, separate from real projects, so you can try things out without fear of losing any code you write. For example:
```bash
cd ~/Documents
mkdir git-testing
cd git-testing
git init
echo "This is for testing git" >> README.md
git add README.md
git commit
# and test out more git commands such as branching and rebasing
```

The default `git log` command isn't helpful for visualizing the repository so I'd recommend running `git log --pretty=oneline --abbrev-commit --graph --decorate --color --all` to better see all your branches and commits. You can create a ~/.gitconfig and alias this too, so you don't have to type it every time, [like this](https://github.com/mcomella/dotfiles/blob/master/git/gitconfig#L17).

> because I can't scroll up without refreshing the page. The version I download off the Play Store doesn't have that problem so I know I'm building the app wrong but I can't figure out how to fix that.

I think your build is working correctly! We release new changes every two weeks so the build you have locally is more up-to-date than what's on Google Play. We added scroll to refresh in #1418, which will come out in the next release.

> So is my next step to just try to figure out how to merge the master branch into my fork without messing everything up? I'm not sure my git is set up correctly that I can get things from the source github...

Github forces you to work with three repositories:
- Your local repositories
- Your online fork (where you probably initially cloned from and where you push your PRs to)
- The original upstream repository (mozilla-mobile/...)

You'll need to add our upstream repository as a remote and `git fetch`/`git pull` from there. It looks like this guide might help: https://help.github.com/articles/syncing-a-fork/

Please let us know if you have any more questions!

**Comment by liuche** [comment]
Please make sure code runs next time before opening a PR!

**Comment by pocmo** [comment]
> Please make sure code runs next time before opening a PR!

I noticed that we run the APK size check only on Buddybuild. Unfortunately we do not run buddybuild for PRs (at least for new contributors). I'll file a bug to add this to the taskcluster build too.
